altair/utils/core.py

```diff
@@ -193,8 +193,6 @@ def infer_vegalite_type(data):
     # Otherwise, infer based on the dtype of the input
     typ = infer_dtype(data)
 
-    # TODO: Once this returns 'O', please update test_select_x and test_select_y in test_api.py
```

Comment:
I assume this refers to ordinal data, but I could not find these specific tests, but I updated the one test that seemed relevant.